src/coredata.cs

```diff
@@ -1166,36 +1462,58 @@ public partial interface NSPersistentStoreCoordinator
 
-		[Static, Export ("registerStoreClass:forStoreType:")]
+		[Export ("registerStoreClass:forStoreType:")]
 		void RegisterStoreClass (Class storeClass, NSString storeType);
 
+		[Introduced (PlatformName.iOS, 3, 0, message: "Use a -metadataForPersistentStoreOfType:URL:options:error: and pass in an options dictionary matching addPersistentStoreWithType")]
+		[Deprecated (PlatformName.iOS, 9, 0, message: "Use a -metadataForPersistentStoreOfType:URL:options:error: and pass in an options dictionary matching addPersistentStoreWithType")]
 		[Static, Export ("metadataForPersistentStoreOfType:URL:error:")]
+		[return: NullAllowed]
 		NSDictionary MetadataForPersistentStoreOfType (NSString storeType, NSUrl url, out NSError error);
+		
+		[iOS (7,0)]
+		[Static, Export ("metadataForPersistentStoreOfType:URL:options:error:")]
+		[return: NullAllowed]
+		NSDictionary<NSString, NSObject> MetadataForPersistentStoreOfType (string storeType, NSUrl url, [NullAllowed] NSDictionary options, out NSError error);
 
+		[Introduced (PlatformName.iOS, 3, 0, message: "Use a -setMetadata:forPersistentStoreOfType:URL:options:error: and pass in an options dictionary matching addPersistentStoreWithType")]
+		[Deprecated (PlatformName.iOS, 9, 0, message: "Use a -setMetadata:forPersistentStoreOfType:URL:options:error: and pass in an options dictionary matching addPersistentStoreWithType")]
 		[Static, Export ("setMetadata:forPersistentStoreOfType:URL:error:")]
 		bool SetMetadata (NSDictionary metadata, NSString storeType, NSUrl url, out NSError error);
+		
+		[iOS (7,0)]
+		[Static, Export ("setMetadata:forPersistentStoreOfType:URL:options:error:")]
+		bool SetMetadata ([NullAllowed] NSDictionary<NSString, NSObject> metadata, string storeType, NSUrl url, [NullAllowed] NSDictionary options, out NSError error);
 
 		[Export ("setMetadata:forPersistentStore:")]
-		void SetMetadata (NSDictionary metadata, NSPersistentStore store);
+#if XAMCORE_4_0
+		void SetMetadata ([NullAllowed] NSDictionary<NSString, NSObject> metadata, NSPersistentStore store);
+#else
+		void SetMetadata ([NullAllowed] NSDictionary metadata, NSPersistentStore store);
+#endif
 
 		[Export ("metadataForPersistentStore:")]
+#if XAMCORE_4_0
+		NSDictionary<NSString, NSObject> MetadataForPersistentStore (NSPersistentStore store);
+#else
 		NSDictionary MetadataForPersistentStore (NSPersistentStore store);
+#endif
 
 		[DesignatedInitializer]
 		[Export ("initWithManagedObjectModel:")]
 		IntPtr Constructor (NSManagedObjectModel model);
 
-		[Export ("managedObjectModel")]
+		[Export ("managedObjectModel", ArgumentSemantic.Strong)]
 		NSManagedObjectModel ManagedObjectModel { get; }
 
-		[Export ("persistentStores")]
+		[Export ("persistentStores", ArgumentSemantic.Strong)]
 		NSPersistentStore[] PersistentStores { get; }
 
 		[Export ("persistentStoreForURL:")]
-		NSPersistentStore PersistentStoreForUrl (NSUrl URL);
+		NSPersistentStore PersistentStoreForUrl (NSUrl Url);
```

Comment:
technically a breaking change (but one we have not really cared so far). Should be changed to `url` (no cap)